sdk/go/pulumi/context.go

```diff
@@ -364,7 +364,7 @@ func (ctx *Context) Call(tok string, args Input, output Output, self Resource, o
 		// If we have a value for self, add it to the arguments.
 		if self != nil {
 			var deps []URN
-			resolvedSelf, selfDeps, err := marshalInput(self, nil, true)
+			resolvedSelf, selfDeps, err := marshalInput(self, anyType, true)
```

Comment:
Feels like we ought to have a better type than `anyType` for `resolvedSelf`. Is there a reason that we can't use the type of the resource itself?

Reply:
Ah, fair point. It doesn't really make a difference here and I was just following [some](https://github.com/pulumi/pulumi/blob/ece9f2fb30598ed4796214f2c91290f31a63a742/sdk/go/pulumi/context.go#L246) [other](https://github.com/pulumi/pulumi/blob/ece9f2fb30598ed4796214f2c91290f31a63a742/sdk/go/pulumi/context.go#L1400) [cases](https://github.com/pulumi/pulumi/blob/ece9f2fb30598ed4796214f2c91290f31a63a742/sdk/go/pulumi/rpc.go#L261) where we use `anyType` with `marshalInput` when the type doesn't matter, but I don't see any reason we can't use the type of the resource itself here. Fixed.